pkg/sql/vecindex/cspann/partition_metadata.go

```diff
@@ -102,6 +106,10 @@ const (
 	// vectors to target split sub-partitions. Searches are allowed, but not
 	// inserts, deletes, splits, or merges.
 	DrainingForSplitState
+	// DeletingForSplitState indicates that a non-root splitting partition is
+	// about to be removed from its tree and deleted. Searches are allowed, but
+	// not inserts, deletes, splits, or merges.
```

Comment:
The partition is already cleared at this point - why allow searches?

Reply:
We always allow searches, no matter what the state is. It's simple and effective. Disabling searches would be more special cases in already complex code for very little benefit.